ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_program_factory.cpp

```diff
@@ -42,20 +42,19 @@ operation::ProgramWithCallbacks pad_rm_reader_writer(const Tensor &a,
             .to(device, MemoryConfig{.memory_layout = TensorMemoryLayout::INTERLEAVED, .buffer_type = BufferType::L1});
     auto pad_value_const_tensor_addr = pad_value_const_tensor.buffer()->address();
 
+    Buffer *src0_buffer = a.buffer();
+    Buffer *dst_buffer = output.buffer();
+    TT_ASSERT(dst_buffer != nullptr, "Output buffer should be allocated on device!");
```

Comment:
shouldn't this be TT_FATAL?

Reply:
these were copied from below and just moved up, should I change it to a fatal?

Reply:
Yeah, I think we should since this is something we want to fire in release builds too. I can make this change after you've merged the PR if you'd prefer.